### PR TITLE
Improved support for custom image loaders

### PR DESCRIPTION
**Pull request by supagu**
The fixup for #19374 which I messed up but have cleaned up here.

Removed image loader limit. Support to add/remove/find/list image loaders. Resolves both #19322 and #19037

@reduz can you check this now. Ta

### CODE REVIEW

File: core/io/image_loader.cpp [hunk 1]
```diff
@@ -82,30 +82,45 @@ Error ImageLoader::load_image(String p_file, Ref<Image> p_image, FileAccess *p_c
...
+const Vector<ImageFormatLoader *> &ImageLoader::get_image_format_loaders() {
+
+	return loader;
+}
+
+void ImageLoader::cleanup() {
+
+    while (loader.size()) {
+        remove_image_format_loader(loader[0]);
+    }
```
Comment (LikeLakers2) [inline core/io/image_loader.cpp]: Indentation should be tab-based, not space-based, per the code style guidelines. http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

File: core/io/image_loader.cpp [hunk 2]
```diff
@@ -137,7 +152,7 @@ RES ResourceFormatLoaderImage::load(const String &p_path, const String &p_origin
 
 	int idx = -1;
 
-	for (int i = 0; i < ImageLoader::loader_count; i++) {
+    for (int i = 0; i < ImageLoader::loader.size(); i++) {
```
Comment (LikeLakers2) [inline core/io/image_loader.cpp]: Indentation again.

File: main/main.cpp [hunk 3]
```diff
@@ -1929,6 +1930,8 @@ void Main::cleanup() {
 		memdelete(arvr_server);
 	}
 
+    ImageLoader::cleanup();
+
```
Comment (LikeLakers2) [inline main/main.cpp]: As well as here.

**Comment by supagu** [comment]
Thanks @LikeLakers2, I forgot to come back and check everything was mergable. Should all be good now.

File: main/main.cpp [hunk 4]
```diff
@@ -1929,6 +1930,8 @@ void Main::cleanup() {
 		memdelete(arvr_server);
 	}
 
+	ImageLoader::cleanup();
```
Comment (reduz) [inline main/main.cpp]: I think this should probably work better in unregister_core_types() function

**Review by reduz — CHANGES_REQUESTED** [review]

File: main/main.cpp (hunk 4 above)
Comment (supagu) [reply main/main.cpp]: It needs to occur before unregister_module_types which frees the image loaders. Moving it to unregister_core_types leaves dangling pointers in the ImageLoader array.

**Comment by reduz** [comment]
Ah, I forgot about this, and it was much better than my fix. Can you please rebase it/fix it, so I can merge it?

**Comment by supagu** [comment]
@reduz Rebased all ready for you to merge!

**Comment by reduz** [comment]
We already discussed 16 PNG support and rejected it, sorry. The reason is that users have no idea what are they saving their image as, and both applications and converters don't really tell you much about this.. the result is huge exports without the users noticing.

**Comment by supagu** [comment]
Woops! Sorry, I didnt see that go in with my rebase. I'll remove it

**Pull request merged**